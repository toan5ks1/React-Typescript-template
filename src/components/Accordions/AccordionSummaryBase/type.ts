import { ReactNode } from "react";

import { IconButtonBaseProps } from "src/components/IconButton/IconButtonBase";

export interface AccordionSummaryIconExpandProps {
    summaryContent: ReactNode;
    avatar?: ReactNode;
    active?: boolean;
    iconButtonProps?: IconButtonBaseProps;
}
