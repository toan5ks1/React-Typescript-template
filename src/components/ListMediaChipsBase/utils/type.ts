import { StringKeyOf } from "ts-enum-util/src/types";

export enum MediaType {
    MEDIA_TYPE_NONE = 0,
    MEDIA_TYPE_VIDEO = 1,
    MEDIA_TYPE_IMAGE = 2,
    MEDIA_TYPE_PDF = 3,
    MEDIA_TYPE_AUDIO = 4,
}

export enum ConversionTaskStatus {
    CONVERSION_TASK_STATUS_INVALID = 0,
    CONVERSION_TASK_STATUS_WAITING = 1,
    CONVERSION_TASK_STATUS_CONVERTING = 2,
    CONVERSION_TASK_STATUS_FINISHED = 3,
    CONVERSION_TASK_STATUS_FAILED = 4,
}

export interface Media {
    comments?: string;
    conversion_tasks: any[];
    conversion_tasks_aggregate: any;
    converted_images?: string;
    created_at: Date;
    deleted_at?: Date;
    media_id: string;
    name?: string;
    resource?: string;
    resource_path?: string;
    type?: string;
    updated_at: Date;
}

export type UploadFilesType<T extends any = any> = File | Media | T;

export const isMedia = (media: UploadFilesType): media is Media => {
    return (media as Media).media_id !== undefined;
};

export function getKeyByValue<T extends Record<StringKeyOf<T>, number | string>>(
    enumObj: T,
    enumValue: string | number
): string | null {
    let keys = Object.keys(enumObj).filter((x) => enumObj[x] == enumValue);
    return keys.length > 0 ? keys[0] : null;
}
