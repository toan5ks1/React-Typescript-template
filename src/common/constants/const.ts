import {
    MediaType,
    TopicStatus,
    TopicType,
    UserGroup,
    UserNotificationStatus,
} from "manabie-bob/enum_pb";
import { QuizOptionConfig } from "manabie-yasuo/quiz_pb";
import { MediaType as MediaTypeV1 } from "manabuf/bob/v1/enums_pb";
import { ConversionTaskStatus } from "manabuf/bob/v1/media_pb";
import { LessonStatus } from "manabuf/common/v1/contents_pb";
import { Country, LearningObjectiveType, Subject } from "manabuf/common/v1/enums_pb";
import {
    NotificationEvent,
    NotificationStatus,
    NotificationTargetGroupSelect,
    NotificationType,
} from "manabuf/common/v1/notifications_pb";
import { StudyPlanTaskStatus, StudyPlanType } from "manabuf/eureka/v1/enums_pb";

import { convertEnumKeys } from "./helper";

export enum ExtendedUserGroup {
    USER_GROUP_HQ_STAFF = 16,
    USER_GROUP_TEACHER_LEAD = 17,
    USER_GROUP_CENTRE_LEAD = 18,
    USER_GROUP_CENTRE_MANAGER = 19,
    USER_GROUP_CENTRE_STAFF = 20,
}

export const UserGroupUnion = Object.assign({}, UserGroup, ExtendedUserGroup);
export const KeyStudyPlanTypes = convertEnumKeys(StudyPlanType);
export const UserRoles = convertEnumKeys(UserGroupUnion); // this is used frequently, so I dont change it naming now
export const KeyMediaTypes = convertEnumKeys(MediaType);
export const KeyMediaTypesV1 = convertEnumKeys(MediaTypeV1);
export const KeyLOTypes = convertEnumKeys(LearningObjectiveType);
export const KeyQuizOptionConfigs = convertEnumKeys(QuizOptionConfig);
export const KeyStudyPlanTaskStatusTypes = convertEnumKeys(StudyPlanTaskStatus);
export const KeyConversionTaskStatusTypes = convertEnumKeys(ConversionTaskStatus);
export const KeyNotificationType = convertEnumKeys(NotificationType);
export const KeyNotificationTargetGroupSelect = convertEnumKeys(NotificationTargetGroupSelect);
export const KeyNotificationStatus = convertEnumKeys(NotificationStatus);
export const KeyNotificationEvent = convertEnumKeys(NotificationEvent);
export const KeyUserNotificationStatus = convertEnumKeys(UserNotificationStatus);

export const KeyTopicTypes = convertEnumKeys(TopicType);
export const KeyTopicStatuses = convertEnumKeys(TopicStatus);

export const KeyCountries = convertEnumKeys(Country);
export const KeySubjects = convertEnumKeys(Subject);
export const KeyLiveLessonStatus = convertEnumKeys(LessonStatus);

export const REGEX_VALIDATE_EMAIL =
    /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

export const COUNTRY_INFO = {
    [Country.COUNTRY_JP]: {
        code: "+81",
        countryCode: "JP",
    },
    [Country.COUNTRY_VN]: {
        code: "+84",
        countryCode: "VN",
    },
};

export const KEY_CODE = {
    KEY_0: 48,
    KEY_9: 57,
    KEY_BACKSPACE: 8,
    KEY_ENTER: 13,
};

export const StudentKeys = {
    RELATIONSHIP: "relationship",
    STUDENTS_STATUS: "studentsEnrollmentStatus",
    ADD_PARENT: "addParent",
};

export const DEFAULT_DEBOUNCE_AUTOCOMPLETE_TIME = 200;
export const DEFAULT_MAX_SNACK = 2;

export const mappedRoles = {
    ["Teacher"]: UserRoles.USER_GROUP_TEACHER,
    ["School Admin"]: UserRoles.USER_GROUP_SCHOOL_ADMIN,
    ["HQ Staff"]: UserRoles.USER_GROUP_HQ_STAFF,
    ["Teacher Lead"]: UserRoles.USER_GROUP_TEACHER_LEAD,
    ["Centre Lead"]: UserRoles.USER_GROUP_CENTRE_LEAD,
    ["Centre Manager"]: UserRoles.USER_GROUP_CENTRE_MANAGER,
    ["Centre Staff"]: UserRoles.USER_GROUP_CENTRE_STAFF,
};
