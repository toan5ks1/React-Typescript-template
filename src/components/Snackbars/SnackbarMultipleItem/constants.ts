export enum CloseReason {
    TIMEOUT = "timeout",
    MAXSNACK = "maxsnack",
    INSTRUCTED = "instructed",
}

export enum DefaultsSnackbar {
    MAX_SNACK = 3,
    VARIANT = "success",
    AUTO_HIDE_DURATION = 5000,
}
