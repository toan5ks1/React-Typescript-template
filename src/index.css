body {
    margin: 0;
    font-family: "Roboto", sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    overflow-x: hidden;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
}

body,
html,
#root {
    margin: 0;
    padding: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
}

#root > div:first-child {
    margin: 0;
    padding: 0;
    width: 100%;
    min-height: 100vh;
    box-sizing: border-box;
}

.list-page > div > div:last-child {
    overflow: auto;
}

table td {
    max-width: 300px;
    overflow-wrap: break-word;
}

/* Override css background input autocomplete - Designer wants it */
input:-webkit-autofill,
input:-webkit-autofill:hover,
input:-webkit-autofill:focus,
textarea:-webkit-autofill,
textarea:-webkit-autofill:hover,
textarea:-webkit-autofill:focus,
select:-webkit-autofill,
select:-webkit-autofill:hover,
select:-webkit-autofill:focus {
    -webkit-box-shadow: 0 0 0px 1000px #fff inset;
    transition: background-color 5000s ease-in-out 0s;
}

/* styled not affectted, so we use this, check src/components/BrightcoveVideo/BrightcoveVideo.tsx */
.bc-player-default_default {
    width: 100%;
    height: 100%;
}
