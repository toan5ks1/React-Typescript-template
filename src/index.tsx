import "src/styles/mui-classname-setup";

import { StrictMode } from "react";

import { enableMapSet } from "immer";
import ReactDOM from "react-dom";
import enableSidebarMethod from "src/squads/architecture/internals/sidebar";
import enableManaEventEmitter from "src/squads/root/src/internals/mana-event-emitter";

import App from "./app";
import AppProvider from "./app/AppProvider";
import "./index.css";
import { enableReport } from "./internals/stackdriver";
import { enableLogger } from "./internals/warner";

enableMapSet();
enableReport();
enableLogger();

enableSidebarMethod();
enableManaEventEmitter();

ReactDOM.render(
    <StrictMode>
        <AppProvider>
            <App />
        </AppProvider>
    </StrictMode>,
    document.getElementById("root")
);
