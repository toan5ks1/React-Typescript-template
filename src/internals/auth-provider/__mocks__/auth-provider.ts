export default {
    __esModule: true,
    default: {
        checkAuth: jest.fn(),
        login: jest.fn(),
        checkError: jest.fn(),
        logout: jest.fn(),
        getPermissions: jest.fn(),
    },
};
