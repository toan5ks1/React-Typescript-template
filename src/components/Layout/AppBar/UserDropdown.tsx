import { useState } from "react";

import { UserIdentity } from "src/typings/auth-provider";

import ArrowDropDown from "@mui/icons-material/ArrowDropDown";
import ArrowDropUp from "@mui/icons-material/ArrowDropUp";
import { styled } from "@mui/material/styles";
import UserInfo from "src/components/RelatedUser/UserInfo";
import UserMenu from "src/components/RelatedUser/UserMenu";

import useLogout from "src/squads/user/hooks/auth/useLogout";

const StyledUserInfo = styled(UserInfo)({
    paddingRight: 0,
});
StyledUserInfo.displayName = "StyledUserInfo";

export interface UserDropdownProps {
    userProfile: UserIdentity;
    onChangeLocationSetting: (() => void) | undefined;
}

const UserDropdown = ({ userProfile, onChangeLocationSetting }: UserDropdownProps) => {
    const { onLogout } = useLogout();
    const [expandUser, setExpandUser] = useState(false);
    return (
        <UserMenu
            onLogout={onLogout}
            onOpenCallBack={(isOpen) => {
                setExpandUser(isOpen);
            }}
            onOpenLocationSetting={onChangeLocationSetting}
        >
            <StyledUserInfo
                data-testid="Appbar__profileButton"
                profile={userProfile}
                icon={expandUser ? ArrowDropUp : ArrowDropDown}
            />
        </UserMenu>
    );
};

export default UserDropdown;
