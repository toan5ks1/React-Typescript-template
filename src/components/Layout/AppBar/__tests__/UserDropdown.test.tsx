import { UserRoles } from "src/common/constants/const";
import { AppProvider, TestApp } from "src/test-utils";
import { UserIdentity } from "src/typings/auth-provider";

import UserDropdown from "src/components/Layout/AppBar/UserDropdown";

import { Country } from "manabie-bob/enum_pb";

import { screen, render, RenderResult, fireEvent } from "@testing-library/react";
import type { GlobalLocationTypesReturn } from "src/hooks/useGlobalLocationTypesWithLocations/useGlobalLocationTypesWithLocations";
import useLogout from "src/squads/user/hooks/auth/useLogout";

const userProfile: UserIdentity = {
    name: "Name",
    id: "id",
    email: "email@domain.com",
    schoolId: 123,
    schoolName: "School",
    countryName: "Vietnam",
    userGroup: UserRoles.USER_GROUP_SCHOOL_ADMIN,
    country: Country.COUNTRY_VN,
    phoneNumber: "",
    avatar: "",
    deviceToken: "",
    schoolIdsList: [],
    schoolsList: [],
};

jest.mock("src/squads/user/hooks/auth/useLogout", () => ({
    __esModule: true,
    default: jest.fn(),
}));

jest.mock("src/hooks/useGlobalLocationTypesWithLocations", () => ({
    __esModule: true,
    default: () => {
        const result: GlobalLocationTypesReturn = {
            sortedLocationTypesWithLocations: [],
            setLocationTypesWithLocations: jest.fn(),
        };
        return result;
    },
}));

describe("<UserDropdown />", () => {
    const changeSettingMockFn = jest.fn();
    const onLogout = jest.fn();
    let wrapper: RenderResult;

    const props = {
        userProfile,
        onChangeLocationSetting: changeSettingMockFn,
    };

    beforeEach(() => {
        (useLogout as jest.Mock).mockImplementation(() => {
            return { onLogout };
        });

        wrapper = render(
            <TestApp>
                <AppProvider>
                    <UserDropdown {...props} />
                </AppProvider>
            </TestApp>
        );
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should render menu popper and 2 menu items when clicking the menu", () => {
        const profileButton = screen.getByTestId("Appbar__profileButton");
        expect(profileButton).toBeInTheDocument();

        fireEvent.click(profileButton);
        expect(screen.getByTestId("UserMenu__popper")).toBeInTheDocument();
        expect(screen.getByTestId("UserMenu__setting")).toBeInTheDocument();
        expect(screen.getByTestId("UserMenu__logout")).toBeInTheDocument();

        fireEvent.click(screen.getByTestId("UserMenu__setting"));
        expect(changeSettingMockFn).toBeCalled();

        fireEvent.click(screen.getByTestId("UserMenu__logout"));
        expect(onLogout).toBeCalled();
    });
});
