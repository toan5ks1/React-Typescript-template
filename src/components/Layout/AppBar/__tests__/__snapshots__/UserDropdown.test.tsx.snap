// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDropdown /> should match snapshot 1`] = `
.emotion-0 .groupProfile {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  color: #212121;
  font-weight: 500;
  font-size: 14px;
  text-transform: unset;
  cursor: pointer;
  margin-left: 32px;
  border: none;
}

.emotion-0 .avatar {
  margin-right: 8px;
  width: 32px;
  height: 32px;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 0.875rem;
  font-weight: 500;
  letter-spacing: 0.4px;
  text-transform: capitalize;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #E0E0E0;
  border-radius: 4px;
  color: #212121;
  padding-right: 0;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(224, 224, 224, 0.04);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-disabled {
  color: #BDBDBD;
}

.emotion-1:hover {
  background-color: rgba(33, 33, 33, 0.04);
}

.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: Roboto,sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fff;
  background-color: #BDBDBD;
}

.emotion-3 {
  margin-right: 4px;
  text-align: left;
}

.emotion-4 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
}

.emotion-5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  color: rgba(0, 0, 0, 0.54);
}

.emotion-6 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div>
    <span
      class="emotion-0"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textDefault MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root groupProfile emotion-1"
        data-testid="Appbar__profileButton"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar emotion-2"
        >
          N
        </div>
        <div
          class="MuiBox-root emotion-3"
          data-testid="Appbar__role"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-4"
          >
            School Admin of School
          </p>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium emotion-5"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root emotion-6"
        />
      </button>
    </span>
  </div>
</div>
`;
